apps/prairielearn/elements/pl-excalidraw/pl-excalidraw.py

```diff
@@ -0,0 +1,218 @@
+        raise RuntimeError(f"Missing required attribute `{ATTR_SOURCE_FILE_NAME}`")
+
+    initial_content: str = empty_diagram
+
+    # We sometimes want to render errors without rendering the widget
+    show_widget = True
+
+    # Notes:
+    # 1. We could have simplified the if-else chains below by collapsing branches, but we use
+    # the expanded form to make it easy to compare with the reference logic at
+    # https://github.com/PrairieLearn/PrairieLearn/pull/9900#discussion_r1685163461
+    # 2. Because we already handled raising an error on missing source file, the error cases
+    # and invalid states are both tagged below as `unreachable`
+    unreachable = RuntimeError(f"Unreachable code path reached with state {matrix=}")
+
+    match panel:
+        case "question":
+            if gradable:
+                if fresh:
+                    if source_available:
+                        initial_content = load_file_content(element, data)
+                    else:
+                        initial_content = empty_diagram
+                else:  # submission
+                    initial_content = data["submitted_answers"][drawing_name]
+            else:  # not gradable
+                if fresh:
+                    if source_available:
+                        initial_content = load_file_content(element, data)
+                    else:
+                        raise unreachable
+                else:  # submission
+                    raise unreachable
+
+        case "answer":
+            if gradable:
+                show_widget = False
+            else:  # not gradable
+                if source_available:
+                    if fresh:
+                        initial_content = load_file_content(element, data)
+                    else:  # submission
+                        raise unreachable
+                else:  # no source file
+                    raise unreachable
+
+        case "submission":
+            if gradable:
+                if fresh:
+                    raise unreachable
+                else:  # submission
+                    initial_content = data["submitted_answers"][drawing_name]
+            else:  # not gradable
+                if fresh:
+                    if source_available:
+                        initial_content = load_file_content(element, data)
+                    else:  # no source file
+                        raise unreachable
+                else:  # submission
+                    raise unreachable
+
+        case panel:
+            assert_never(panel)
+
+    content_bytes = json.dumps(
+        {
+            "read_only": not is_widget_editable(panel, gradable, data["editable"]),
+            "initial_content": initial_content,
+            "width": pl.get_string_attrib(element, ATTR_WIDTH, "100%"),
+            "height": pl.get_string_attrib(element, ATTR_HEIGHT, "800px"),
+        }
+    ).encode()
+
+    errors: List = []
```

Comment:
```suggestion
    errors: list = []
```
use newer lowercase annotations